server/src/main/java/org/elasticsearch/search/aggregations/support/ValuesSourceAggregationBuilder.java

```diff
@@ -347,16 +349,10 @@ public ZoneId timeZone() {
     protected final ValuesSourceAggregatorFactory doBuild(AggregationContext context, AggregatorFactory parent,
                                                           Builder subFactoriesBuilder) throws IOException {
         ValuesSourceConfig config = resolveConfig(context);
-        if (context.getValuesSourceRegistry().isRegistered(getRegistryKey())) {
-            /*
-            if the aggregation uses the values source registry, test if the resolved values source type is compatible with this aggregation.
-            This call will throw if the mapping isn't registered, which is what we want.  Note that we need to throw from here because
-            AbstractAggregationBuilder#build, which called this, will attempt to register the agg usage next, and if the usage is invalid
-            that will fail with a weird error.
-             */
```

Comment:
I'd like to leave some version of this comment; I find it useful to note that this is our last chance to check the values source type mapping before we attempt to register usage.  Maybe something like 

```
/*
The inner builder implementation is responsible for validating the 
ValuesSourceType mapping, typically by checking if an aggregation
supplier has been registered for that type on this aggregation, and
throw IllegalArgumentException if the mapping is not valid.  Note 
that we need to throw from here because 
AbstractAggregationBuilder#build, which called this, will attempt to
register the agg usage next, and if the usage is invalid that will fail
with a weird error.
*/
```